codalab/lib/worksheet_util.py

```diff
@@ -457,6 +457,11 @@ def apply_func(func, arg):
                     arg = arg[start:end]
                 else:
                     return '<invalid function: %s>' % f
+            elif f.startswith('add'):
```

Comment:
for 'add' and 'key', check that arg is actually a hash table
also comment on what 'add' and 'key' do